File: examples/dart/receiver.dart
// --- Receiver ---
import 'dart:io';
import 'dart:convert';

void main() async {
  final socket = await RawDatagramSocket.bind(InternetAddress.anyIPv4, 8080);
  print('Listening on ${socket.address.address}:${socket.port}...');

  await for (final event in socket) {
    if (event == RawSocketEvent.read) {
      final datagram = socket.receive();
      if (datagram == null) continue;
      final message = utf8.decode(datagram.data);
      print("Received '$message' from ${datagram.address.address}:${datagram.port}");
      socket.send(utf8.encode('Message received'), datagram.address, datagram.port);
    }
  }
}

File: examples/javascript/receiver.js
// --- Receiver ---
const dgram = require('dgram');
const server = dgram.createSocket('udp4');
server.on('message', (msg, rinfo) => {
  console.log(`Received '${msg}' from ${rinfo.address}:${rinfo.port}`);
  server.send('Message received', rinfo.port, rinfo.address);
});
server.on('listening', () => console.log(`Listening on ${server.address().port}...`));
server.bind(8080);
